zebrad/tests/acceptance.rs

```diff
@@ -351,3 +351,61 @@ fn valid_generated_config(command: &str, expected_output: &str) -> Result<()> {
 
     Ok(())
 }
+
+#[test]
+fn listener_address_test() -> Result<()> {
+    listener_address(
+        "localhost:8233",
+        Some("Opened Zcash protocol endpoint at 127.0.0.1:8233"),
+    )?;
+    listener_address(
+        "127.0.0.1:8233",
+        Some("Opened Zcash protocol endpoint at 127.0.0.1:8233"),
+    )?;
+    listener_address(
+        "::1:8233",
+        Some("Opened Zcash protocol endpoint at \\[::1\\]:8233"),
+    )?;
+    listener_address("invalid", None)?;
+
+    Ok(())
+}
+
+fn listener_address(request: &str, expected_response: Option<&str>) -> Result<()> {
+    zebra_test::init();
+
+    // Create a temp dir to work in
+    let dir = TempDir::new("zebrad_tests")?;
+
+    // Create a config file with request as listen_addr
+    let mut config = ZebradConfig::default();
+    config.state.cache_dir = dir.path().join("state");
+    config.state.memory_cache_bytes = 256000000;
```

Comment:
How can we make sure that this config doesn't get out of sync with the default test config?

Should we have a function that returns the default test config?

Reply:
Added function at https://github.com/ZcashFoundation/zebra/pull/951/commits/ba0e42e930c60740d86210f602c89205762d58c8